miden-lib/asm/sat/note.masm

```diff
@@ -0,0 +1,34 @@
+use.miden::sat::layout
+
+#! Returns the sender of the note currently being processed. If we are not in the transaction
+#! prologue then ZERO will be returned.
+#!
+#! Inputs: []
+#! Outputs: [sender]
+#!
+#! - sender is the sender of the note currently being processed.
+export.get_sender
+    # pad with the default return value
+    push.0
+    # => [ZERO]
+
+    # get a flag indicating whether we are in the transaction prologue
+    exec.layout::get_in_prologue
+    # => [in_prologue, ZERO]
+
+    # if we are in the transaction prologue then fetch the sender for the current note.
+    if.true
+        # drop the default value
+        drop
+        # => []
+
+        # fetch the sender for the current note
+        exec.layout::get_current_consumed_note_ptr
+        # => [note_ptr]
+
+        # fetch the sender for the current note
+        exec.layout::get_consumed_note_sender
+    end
+
+    # => [sender]
```

Comment:
Now that I think about it, if we change `get_current_consumed_note_ptr` to return ZERO when not executing a note, we could do this as:
```
exec.layout::get_current_consumed_note_ptr
dup neq.0
if.true
  exec.layout::get_consumed_note_sender
end
```